upgrade_tests/upgrade_through_versions_test.py

```diff
@@ -898,6 +887,22 @@ def create_upgrade_class(clsname, version_metas, protocol_version,
     return newcls
 
 
+def jdk_compatible_steps(version_metas):
+    metas = []
+    java_version = current_env_java_version()
+    for version_meta in version_metas:
+        if java_version in version_meta.java_versions:
+            metas.append(version_meta)
+
+    return metas
+
+
+def current_env_java_version():
+    java_command = os.path.join(os.environ['JAVA_HOME'], 'bin', 'java')
```

Comment:
please fail somewhere with a clear message when `JAVA_HOME` is not defined

Reply:
not needed, see update.